### ISSUE TITLE
BUG: Checkbox current state is not saved in draft on Connect bank account page reported by @parasharrajat 

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11687]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app on web.
2. Go to settings page.
3. Open any workspace(Create new if does not exist).
4. Scroll to the end of workspace settings page.
5. Click Connect bank account menu.
6. Now click Connect manually
7. Type any value on the fields.
8. Check or uncheck the Terms and condition checkbox.
9. Remember the values entered in 7 and 8. 
10. Close the page via clicking :x: icon on top right.
11. Repeat 2-6.
12. Previously entered value in step 7 should be there now.
13. Observe the state of terms and condition checkbox. It is not same as last left in step 8.

## Expected Result:
As the code suggests to save the state of Terms and condition checkbox as draft, it should save it as draft.

## Actual Result:
 Values in text fields are saved in draft but value of terms and condition checkbox is not stored in draft.

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.12-2
**Reproducible in staging?:** y
**Reproducible in production?:**y 
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/194735099-0147ce42-a4c2-4c8a-83a7-9b241cdca3ed.mp4

https://user-images.githubusercontent.com/43996225/194735105-7fbcbe38-3a87-4879-9578-e1d564d4e514.mp4

**Expensify/Expensify Issue URL:** 
**Issue reported by:** @parasharrajat 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1665180050268169

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by muttmuure** [comment]
I'm not sure if we have chosen deliberately to not store the accepted terms and conditions, because we need the customer to agree to the terms at a certain time

**Comment by muttmuure** [comment]
@kavimuru I think we aren’t storing the value in the terms and conditions checkbox as a draft seems to be a deliberate decision we’ve made, and not a bug.

Customers always have to opt into terms & conditions, and, under this proposal, those terms & conditions may have been different at the time we stored the draft which puts us in an awkward position if they don’t like terms of service changes. 

If you disagree, we can discuss in #expensify-open-source or please reopen the issue.

**Comment by parasharrajat** [comment]
@muttmuure I agree with what you said. but there is a bug in the code which prevent any checkboxes from being stored as a draft. I looked at the code and it seems that the code is configured to save the draft for the checkbox as well. But due to the bug, it is not saving the draft value.

I reported this issue to fix that bug. We can leave the terms and conditions checkbox out of the draft but we should fix the bug in the code.

**Comment by muttmuure** [comment]
Copying this point as next steps:

1. fix saving of draft values for checkbox (so any checkbox that wants to save draft, it should work)
2. remove draft value for terms and conditions checkbox

**Comment by parasharrajat** [comment]
@muttmuure I think you forgot to reopen this.

**Comment by amyevans** [comment]
Reproduced, and I agree with the next steps [here](https://github.com/Expensify/App/issues/11687#issuecomment-1283914739). Adding `External`.

**Comment by trjExpensify** [comment]
Job is on Upwork here: https://www.upwork.com/jobs/~011127e7b46973fc12

**Comment by trjExpensify** [comment]
@MonilBhavsar can you review this proposal so @rushatgabhane can move forward with the PR?

**Comment by MonilBhavsar** [comment]
Looks good to me too 👍 

> And remove shouldSaveDraft from checkboxes for terms and conditions

Let's update this for all forms

**Comment by trjExpensify** [comment]
Sent offers to you both on Upwork!

**Comment by parasharrajat** [comment]
Bump @MonilBhavsar for PR review.

**Comment by MonilBhavsar** [comment]
Sorry was OOO and not sure why pullerbear didn't assign the review

**Comment by trjExpensify** [comment]
👋 back from OOO. Nice to see Melv added that checklist three times just in case I forget. 😉 

We're waiting on this to hit production, but to get a start on the checklist.

> The PR that introduced the bug has been identified. Link to the PR:
> The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

@rushatgabhane @parasharrajat was this a regression or just something that was missed during the initial implementation? If the former, do you know what PR introduced it? 

>A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the updated test here:

@isagoico @kavimuru do our existing tests for the bank account flow cover ticking the checkboxes to check for the draft state, or does that need to be updated?

**Comment by parasharrajat** [comment]
This issue was not a regression but a bug that was missed a month back during refactoring.

**Comment by trjExpensify** [comment]
Maybe rephrased, did this ever work in the past?

**Comment by trjExpensify** [comment]
Got it, so this doesn't have a RCA then. It's not a regression/bug (whatever we want to call it), it's simply something we missed when implementing the VBA flow.

**Comment by trjExpensify** [comment]
Settled up! $250 to @rushatgabhane for the fix, $500 to @parasharrajat for reporting & C+. 

>A [regression test](https://expensify.testrail.io/index.php?/dashboard) has been added or updated so that the same bug will not reach production again. Link to the updated test here:

Coming from [here](https://github.com/Expensify/Expensify/issues/232690), @isagoico @kavimuru, I think we need to add the checkboxes to [this test case](https://expensify.testrail.io/index.php?/cases/view/1971143) right?

**Comment by trjExpensify** [comment]
Alrightyy.. I caught up with @isagoico and the [test case](https://expensify.testrail.io/index.php?/cases/view/1971143) has been modified to account for the checkbox in addition to the form field values being saved upon returning. 

Completed the [checklist](https://github.com/Expensify/App/issues/11687#issuecomment-1293290801). We're done here!
